### ISSUE TITLE
[MacOS] TensorFlow Metal Plugin Symbol Not Found Error with TensorFlow 2.18.0

### ISSUE THREAD

**Issue by yuchen001** [tensorflow/tensorflow#81987]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

2.18.0

### Custom code

No

### OS platform and distribution

macOS 15.1.1

### Python version

3.9

### Current behavior?

When trying to import TensorFlow with Metal plugin support on macOS using a virtual environment, an error occurs while loading the Metal plugin. Specifically, TensorFlow fails to locate a required symbol in the shared library libmetal_plugin.dylib. The error indicates a mismatch between the TensorFlow core library and the Metal plugin.

### Standalone code to reproduce the issue

```shell
mkdir test_tf_latest
/opt/homebrew/bin/python3.9 -m venv .venv
source .venv/bin/activate
pip install tensorflow tensorflow-metal

python -c "import tensorflow as tf; print('TensorFlow version:', tf.__version__); print('Num GPUs Available:', len(tf.config.list_physical_devices('GPU')))"
```

### Relevant log output

```shell
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/.../.../test_tf_latest/.venv/lib/python3.9/site-packages/tensorflow/__init__.py", line 437, in <module>
    _ll.load_library(_plugin_dir)
  File "/.../.../test_tf_latest/.venv/lib/python3.9/site-packages/tensorflow/python/framework/load_library.py", line 151, in load_library
    py_tf.TF_LoadLibrary(lib)
tensorflow.python.framework.errors_impl.NotFoundError: dlopen(/.../.../test_tf_latest/.venv/lib/python3.9/site-packages/tensorflow-plugins/libmetal_plugin.dylib, 0x0006): Symbol not found: __ZN3tsl8internal10LogMessageC1EPKcii
  Referenced from: <D2EF42E3-3A7F-39DD-9982-FB6BCDC2853C> /.../.../test_tf_latest/.venv/lib/python3.9/site-packages/tensorflow-plugins/libmetal_plugin.dylib
  Expected in:     <2A053E7E-6DBA-37C2-A28E-1D52A5836870> /.../.../test_tf_latest/.venv/lib/python3.9/site-packages/tensorflow/python/_pywrap_tensorflow_internal.so
```

**Comment by Venkat6871** [comment]
Hi **@yuchen001** ,
Apologies for the delay, and thank you for raising your concern. Instead of using the command `pip install tensorflow tensorflow-metal` , please use the specific compatible version as [documented](https://pypi.org/project/tensorflow-metal/): `pip install tensorflow==2.14.0 tensorflow-metal`. Additionally, the default NumPy version being installed is 2.02, but the compatible version is 1.24.3. You can install it with:`pip install numpy==1.24.3`.
I have tested this setup, and it is detecting everything as we expected.
```
(myenv) maayara-macbookpro:myenv maayara$ python -c "import tensorflow as tf; print('TensorFlow version:', tf.__version__); print('Num GPUs Available:', len(tf.config.list_physical_devices('GPU')))"
TensorFlow version: 2.14.0
Num GPUs Available: 1
(myenv) maayara-macbookpro:myenv maayara$ 
```
Thank you!

**Comment by elvisli** [comment]
I have the same issue.
Have to use the previous version of tensorflow.
Fixed by installing tensorflow==2.17.0 and tensorflow-metal==1.1.0

**Comment by wal5hy** [comment]
This finally fixed my set up thankyou!
Here's the versions which now can run on a Mac M1 Silicon, and successfully reproduce the Image Classification Notebook here: https://www.tensorflow.org/tutorials/images/classification

# Python 3.11.9
keras==2.14.0
numpy==1.24.3
tensorflow==2.14.0
tensorflow-metal==1.1.0

The final piece was numpy version, I cant see this requirement documented anywhere


> Hi **@yuchen001** , Apologies for the delay, and thank you for raising your concern. Instead of using the command `pip install tensorflow tensorflow-metal` , please use the specific compatible version as [documented](https://pypi.org/project/tensorflow-metal/): `pip install tensorflow==2.14.0 tensorflow-metal`. Additionally, the default NumPy version being installed is 2.02, but the compatible version is 1.24.3. You can install it with:`pip install numpy==1.24.3`. I have tested this setup, and it is detecting everything as we expected.
> 
> ```
> (myenv) maayara-macbookpro:myenv maayara$ python -c "import tensorflow as tf; print('TensorFlow version:', tf.__version__); print('Num GPUs Available:', len(tf.config.list_physical_devices('GPU')))"
> TensorFlow version: 2.14.0
> Num GPUs Available: 1
> (myenv) maayara-macbookpro:myenv maayara$ 
> ```
> 
> Thank you!
